android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/home/HomeViewModel.kt

```diff
@@ -24,24 +26,63 @@ class HomeViewModel(
     private val _avatarUrlString = MutableStateFlow("")
     val avatarUrlString: StateFlow<String> = _avatarUrlString.asStateFlow()
 
+    private val _surveysUiModel: MutableStateFlow<HomeSurveysUiModel?> = MutableStateFlow(null)
+    val surveysUiModel: StateFlow<HomeSurveysUiModel?> = _surveysUiModel.asStateFlow()
+
+    private lateinit var surveys: List<Survey>
+    private var currentSurveyIndex: Int = 0
+
     init {
         _currentDate.value = dateTimeFormatter.getFormattedString(
             dateTime.today(),
             DateFormat.WeekDayMonthDay
         ).uppercase()
     }
 
-    fun loadProfile() {
+    fun loadProfileAndSurveys() {
         showLoading()
         viewModelScope.launch {
-            getProfileUseCase()
+            val getProfile = getProfileUseCase()
+            // TODO: Improve this when we have paging implementation
+            val getSurveys = getSurveysUseCase(1, 5)
+            getProfile
+                .combine(getSurveys) { profile, surveys -> Pair(profile, surveys) }
```

Comment:
If your idea is about executing 2 stream parallel, so I think the `merge` operator matches your idea. Please check here: https://kt.academy/article/cc-flow-combine cc @luongvo

Reply:
My idea is to wait until we get all data. So I prefer using the `combine` operator. If we use merge, the type of all flows should be the same.

<img width="674" alt="Screenshot 2022-12-05 at 16 59 28" src="https://user-images.githubusercontent.com/19943832/205608630-5dd6b095-6cfa-4dd9-aae7-c62606a8aaa4.png">

Reply:
@suho what if one of them occurs an error, say it's the surveys data, should we show the user profile info?

Reply:
@luongvo With current logic, it would not show the user profile 🤡

Reply:
> Lol, it's more on the business side 🙈

@suho Yes, as a senior role, I expect you more on this. Sorry mate 😂

Reply:
Fixed in https://github.com/suho/kmm-ic/pull/107/commits/beae7674d690e5854ce56b7a98b4577a2dec5885